src/Leaflet.VectorGrid.js

```diff
@@ -188,15 +193,18 @@ L.VectorGrid = L.GridLayer.extend({
 
 	_updateStyles: function(feat, renderer, styleOptions) {
 		styleOptions = (styleOptions instanceof Function) ?
-			styleOptions(feat.properties, renderer.getCoord().z) :
+			styleOptions(feat.properties, renderer.getCoord().z, feat.type) :
 			styleOptions;
 
 		if (!(styleOptions instanceof Array)) {
 			styleOptions = [styleOptions];
 		}
 
 		for (var j = 0; j < styleOptions.length; j++) {
-			var style = L.extend({}, L.Path.prototype.options, styleOptions[j]);
+			var styleOption = (styleOptions[j] instanceof Function) ?
```

Comment:
Again, `styleOption` is a best guess at a reasonable var name. No idea if you have other conventions, such as `thisStyleOption`, `eachStyleOption`.